src/pykoala/corrections/sky.py

```diff
@@ -734,6 +868,211 @@ def combine_telluric_corrections(list_of_telcorr, ref_wavelength):
 
     telluric_correction = np.nanmedian(telluric_corrections, axis=0)
     return TelluricCorrection(telluric_correction=telluric_correction, wavelength=ref_wavelength, verbose=False)
+
+
+# =============================================================================
+# Self-calibration based on strong sky lines
+# =============================================================================
+
+class WaveletFilter(object):
+    '''
+    Estimate overall fibre throughput and wavelength calibration based on sky emission lines (from wavelet transform).
+    
+    Description
+    -----------
+    Given a Row-Stacked Spectra (RSS) object:
+    1. Estimate the FWHM of emission lines from the autocorrelation of the median (~ sky) spectrum.
+    2. Apply a (mexican top hat) wavelet filter to detect features on that scale (i.e. filter out the continuum).
+    3. Find regions actually dominated by sky lines (i.e. exclude lines from the target).
+    4. Estimate fibre throughput from norm (standard deviation).
+    5. Estimate wavelength offset of each fibre (in pixels) from cross-correlation with the sky.
     
+    Attributes
+    ----------
+    - scale: (int)
+        Scale of the wavelet filter, in pixels.
+    - filtered: numpy.ndarray(float)
+        Throughput-corrected wavelet coefficient (filtered rss.intensity).
+    - sky_lo: numpy.ndarray(float)
+        Lower percentile (16) of the estimated filtered sky.
+    - sky: numpy.ndarray(float)
+        Estimated (median) filtered sky.
+    - sky_hi: numpy.ndarray(float)
+        Upper percentile (84) of the estimated filtered sky.
+    - sky_weight: numpy.ndarray(float)
+        Estimated dominance of the sky (between 0 and 1).
+    - fibre_throughput: numpy.ndarray(float)
+        Overall throughput of each fibre.
+    - wavelength: numpy.ndarray(float)
+        Wavelength of the wavelet coefficients (only used for plotting).
+    - fibre_offset: numpy.ndarray(float)
+        Relative wavelenght calibration. Offset, in pixels, with respect to the sky (from weighted cross-correlation).
+    '''
+    
+    def __init__(self, rss):
+        
+        assert isinstance(rss, RSS) == True
```

Comment:
This approach should be adopted in the rest of the modules when checking the DC. Btw, I believe the ` == True` is not necessary.